gpytorch/models/exact_prediction_strategies.py

```diff
@@ -73,6 +76,78 @@ def _exact_predictive_covar_inv_quad_form_root(self, precomputed_cache, test_tra
         # where S S^T = (K_XX + sigma^2 I)^-1
         return test_train_covar.matmul(precomputed_cache)
 
+    def get_fantasy_strategy(self, inputs, targets, full_inputs, full_targets, full_output):
+        full_mean, full_covar = full_output.mean, full_output.lazy_covariance_matrix
+
+        batch_shape = full_inputs[0].shape[:-2]
+
+        full_mean = full_mean.view(*batch_shape, -1)
+        num_train = self.num_train
+
+        # Evaluate fant x train and fant x fant covariance matrices, leave train x train unevaluated.
+        fant_fant_covar = full_covar[..., num_train:, num_train:]
+        mvn = self.likelihood(MultivariateNormal(torch.zeros(1), fant_fant_covar), inputs)
+        fant_fant_covar = mvn.covariance_matrix
+
+        fant_train_covar = full_covar[..., num_train:, :num_train].evaluate()
+        fant_mean = full_mean[..., num_train:]
+
+        self.fantasy_inputs = inputs
+        self.fantasy_targets = targets
+
+        # Get cached K inverse decomp. (or compute if we somehow don't already have the covariance cache)
+        K_inverse = self.lik_train_train_covar.root_inv_decomposition()
+        fant_solve = K_inverse.matmul(fant_train_covar.transpose(-2, -1))
+
+        schur_complement = fant_fant_covar - fant_train_covar.matmul(fant_solve)
+        small_system_rhs = targets - fant_mean - fant_train_covar.matmul(self.mean_cache)
+
+        # Schur complement of a spd matrix is guaranteed to be positive definite
+        if small_system_rhs.requires_grad or schur_complement.requires_grad:
+            # TODO: Delete this part of the if statement when PyTorch implements potrs derivative.
```

Comment:
Is there an issue open? If so let's add this to the comment.

Reply:
No issue open as far as I know